pkg/cri/io/logger.go

```diff
@@ -171,7 +203,7 @@ func redirectLogs(path string, rc io.ReadCloser, w io.Writer, s StreamType, maxL
 				panic("exceed length should <= last buffer size")
 			}
 			buf[len(buf)-1] = last[:len(last)-exceedLen]
-			writeLine(partial, bytes.Join(buf, nil))
+			writeLogWithLimiter(partial, bytes.Join(buf, nil))
```

Comment:
i would be happier if we check for the burst/limit here and not rely on setting the rateLimiter to `Inf` in (`rate.NewLimiter(rate.Inf, 0)`)

```
if logBurst > 0 && logLimit > 0 {
    writeLogWithLimiter(partial, bytes.Join(buf, nil))
} else {
    writeLine(partial, bytes.Join(buf, nil))
}
```

Reply:
Thanks, I will rewrite this part